packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -0,0 +1,1459 @@
+     */
+    event KroUndelegationFinalized(
+        address indexed validator,
+        address indexed delegator,
+        uint128 amount
+    );
+
+    /**
+     * @notice Emitted when KGH undelegation is finalized.
+     *
+     * @param validator Address of the validator.
+     * @param delegator Address of the delegator.
+     * @param amount    The amount of KRO undelegated during the KGH undelegation.
+     */
+    event KghUndelegationFinalized(
+        address indexed validator,
+        address indexed delegator,
+        uint128 amount
+    );
+
+    /**
+     * @notice Emitted when pending challenge reward for challenge winner is distributed.
+     *
+     * @param recipient Address of the reward recipient.
+     * @param amount    The amount of challenge reward.
+     */
+    event ChallengeRewardDistributed(address recipient, uint128 amount);
+
+    /**
+     * @notice Emitted when the output reward is distributed.
+     *
+     * @param recipient Address of the reward recipient.
+     * @param amount    The amount of reward.
+     */
+    event RewardDistributed(address recipient, uint128 amount);
+
+    /**
+     * @notice Emitted when the validator is slashed.
+     *
+     * @param loser  Address of the loser at the challenge.
+     * @param winner Address of the winner at the challenge.
+     * @param amount The amount of KRO slashed.
+     */
+    event Slashed(address indexed loser, address indexed winner, uint128 amount);
+
+    /**
+     * @notice Modifier to check if the caller is the Colosseum contract.
+     */
+    modifier onlyColosseum() {
+        require(
+            msg.sender == L2_ORACLE.COLOSSEUM(),
+            "AssetManager: Only Colosseum can call this function"
+        );
+        _;
+    }
+
+    /**
+     * @notice Modifier to check if the vault is active.
+     */
+    modifier checkIsActive(address validator) {
+        require(
+            msg.sender == validator || _vault[validator].asset.validatorKro >= MIN_REGISTER_AMOUNT,
+            "AssetManager: Vault is inactive"
+        );
+        _;
+    }
+
+    /**
+     * @notice Constructs the AssetManager contract.
+     *
+     * @param _l2OutputOracle         Address of the L2OutputOracle contract.
+     * @param _securityCouncil        Address of the SecurityCouncil contract.
+     * @param _token                  Address of the KRO token.
+     * @param _kgh                    Address of the KGH token.
+     * @param _kghManager             Address of the KGHManager contract.
+     * @param _maxRewardRecipients    Max number of reward recipients.
+     * @param _baseReward             Base reward for the validator.
+     * @param _slashingRateNumerator  Numerator of the slashing rate.
+     * @param _minSlashingAmount      Minimum amount to slash.
+     * @param _minRegisterAmount      Minimum amount to register as a validator.
+     * @param _withdrawalDelay        Withdrawal delay for the undelegation.
```

Comment:
Let's make the same order with args of constructor.
And `_minStartAmount` is missing.